insights/parsers/krb5.py

```diff
@@ -0,0 +1,138 @@
+            line = line.strip()
+            if "includedir" in line:
+                continue
+            if is_squ:
+                # If in {} sub_section, get the key_value pair
+                if "=" in line:
+                    key, value = [i.strip() for i in line.split('=', 1)]
+                    if key in squ_value:
+                        if isinstance(squ_value[key], str):
+                            squ_value[key] = [squ_value[key]]
+                        squ_value[key].append(value)
+                    else:
+                        squ_value[key] = value
+                # The {} sub_section should end with }, if it is, set the whole value to the sub_section name, and clean the flag
+                else:
+                    section_value[squ_section_name] = squ_value
+                    is_squ = False
+                    squ_section_name = ""
+                    squ_value = {}
+            else:
+                # [XXX] means a section, get the section name and the value format is dict.
+                if line.startswith("[") and line.endswith("]"):
+                    # If first section, just get the section name, if not, set the value to the former section and get the section name
+                    if section_name:
+                        dict_all[section_name] = section_value
+                    section_name = line.strip("[]")
+                    section_value = {}
+                # key value format is XXX = YYY, store as dict
+                elif "=" in line and not line.endswith("{"):
+                    key, value = [i.strip() for i in line.split('=', 1)]
+                    section_value[key] = value
+                # The {} sub_section should start with format XXX = {
+                else:
+                    is_squ = True
+                    squ_section_name = line.split("=")[0].strip()
+        # Set the value to the last section
+        if section_name and section_value:
+            dict_all[section_name] = section_value
+        self.data = dict_all
+
+    def sections(self):
+        """
+        Return a list of section names.
+        """
+        return self.data.keys()
+
+    def has_section(self, section):
+        """
+        Indicate whether the named section is present in the configuration.
+        Return True if the given section is present, and False if not present.
+        """
+        return section in self.data
+
+    def options(self, section):
+        """
+        Return a list of option names for the given section name.
+        """
+        return self.data[section].keys() if self.has_section(section) else []
+
+    def has_option(self, section, option):
+        """
+        Check for the existence of a given option in a given section.
+        Return True if the given option is present, and False if not present.
+        """
+        if section not in self.data:
+            return False
+        return option in self.data[section]
```

Comment:
This still provides no assistance to rule writers to determine information about the configuration as opposed to merely what the file contained.  How does a rule writer find out what realms are defined? Can they easily find out the ticket lifetime for any domain, with reference back to the defaults? How would a rule get the reverse mapping from realm back to its domains? Etc.  Things like these should be provided as properties or methods to assist the rule writer.

Reply:
@PaulWay - As we need to merge this PR this week for writing a IDM plugin, I plan to write these handy functions in the future.